src/frontend/src/btc/schedulers/btc-wallet.scheduler.ts

```diff
@@ -46,100 +65,159 @@ export class BtcWalletScheduler implements Scheduler<PostMessageDataRequestBtc>
 
 			const newTransactions = fetchedTransactions.filter(({ hash }) =>
 				isNullish(this.store.transactions[`${hash}`])
 			);
 
-			this.store = {
-				...this.store,
-				transactions: {
-					...this.store.transactions,
-					...newTransactions.reduce(
-						(acc, transaction) => ({
-							...acc,
-							[transaction.hash]: {
-								certified: false,
-								data: transaction
-							}
-						}),
-						{}
-					)
-				}
-			};
-
 			const latestBitcoinBlockHeight = await btcLatestBlockHeight();
 
-			return { newTransactions, latestBitcoinBlockHeight };
+			return nonNullish(latestBitcoinBlockHeight)
+				? newTransactions.map((transaction) => ({
+						data: mapBtcTransaction({ transaction, btcAddress, latestBitcoinBlockHeight }),
+						certified: false
+					}))
+				: [];
 		} catch (error) {
 			// We don't want to disrupt the user experience if we can't fetch the transactions or latest block height.
 			console.error('Error fetching BTC transactions data:', error);
-			// TODO: Return an error instead of an object with empty array.
-			return { newTransactions: [], latestBitcoinBlockHeight: undefined };
+			// TODO: Return an error instead of an empty array.
+			return [];
 		}
 	}
 
-	private async loadBtcBalance({
+	private loadBtcBalance = async ({
 		identity,
-		bitcoinNetwork
-	}: {
-		identity: OptionIdentity;
-		bitcoinNetwork: BitcoinNetwork;
-	}): Promise<CertifiedData<bigint>> {
-		const balance = await getBtcBalance({
-			identity,
-			network: bitcoinNetwork,
-			minConfirmations: BTC_BALANCE_MIN_CONFIRMATIONS
-		});
-		const certifiedBalance = {
-			data: balance,
+		bitcoinNetwork,
+		btcAddress,
+		minterCanisterId,
+		certified = true
+	}: Omit<LoadBtcWalletParams, 'shouldFetchTransactions'>): Promise<
+		CertifiedData<bigint> | undefined
+	> => {
+		// Query BTC balance only if minterCanisterId is available
+		if (!certified) {
+			if (nonNullish(minterCanisterId) && isNullish(this.store.balance)) {
+				return {
+					data: await getBalanceQuery({
+						identity,
+						network: bitcoinNetwork,
+						address: btcAddress,
+						bitcoinCanisterId: BITCOIN_CANISTER_IDS[minterCanisterId]
+					}),
+					certified: false
+				};
+			}
+
+			return;
+		}
+
+		return {
+			data: await getBtcBalance({
+				identity,
+				network: mapToSignerBitcoinNetwork({
+					network: bitcoinNetwork
+				}),
+				minConfirmations: BTC_BALANCE_MIN_CONFIRMATIONS
+			}),
 			certified: true
 		};
+	};
 
-		this.store = {
-			...this.store,
-			balance: certifiedBalance
-		};
+	private loadWalletData = async ({
+		certified,
+		identity,
+		bitcoinNetwork,
+		btcAddress,
+		minterCanisterId,
+		shouldFetchTransactions
+	}: LoadBtcWalletParams) => {
+		const balance = await this.loadBtcBalance({
+			identity,
+			bitcoinNetwork,
+			certified,
+			btcAddress,
+			minterCanisterId
+		});
 
-		return certifiedBalance;
-	}
+		// TODO: investigate and implement "update" call for BTC transactions
+		const transactions =
```

Comment:
nitpick: on main we used the name `uncertifiedTransactions`. Maybe it would be nice to use it again for readability reasons

```
// TODO: investigate and implement "update" call for BTC transactions
const. uncertifiedTransactions = ...
return{ balance, transactions: uncertifiedTransactions };

Reply:
Yep, agree, updated.